java/src/main/java/com/google/openlocationcode/OpenLocationCode.java

```diff
@@ -555,34 +556,44 @@ public String toString() {
    * @return True if it is a valid full or short code.
    */
   public static boolean isValidCode(String code) {
+    return toValidCode(code) != null;
+  }
+
+  /**
+   * Normalizes Open Location, or returns `null` if the code is not valid.
+   *
+   * @param code The code to check and normalize.
+   * @return Normalized code if it is a valid full or short code, null otherwise.
+   */
+  private static String toValidCode(String code) {
```

Comment:
I think returning an Optional<String> instead of using null to signify invalid would be slightly cleaner here, what do you think?